pkg/localstorev2/internal/chunkstamp/chunkstamp.go

```diff
@@ -0,0 +1,224 @@
+}
+
+// Namespace implements the storage.Item interface.
+func (i *item) Namespace() string {
+	return path.Join("chunkStamp", i.address.ByteString())
+}
+
+// Marshal implements the storage.Item interface.
+// address is not part of the payload which is stored, as address is part of the
+// prefix, hence already known before querying this object. This will be reused
+// during unmarshaling.
+func (i *item) Marshal() ([]byte, error) {
+	// The address is not part of the payload, but it is used to create the
+	// namespace so it is better if we check that the address is correctly
+	// set here before it is stored in the underlying storage.
+
+	switch {
+	case len(i.namespace) == 0:
+		return nil, errMarshalInvalidChunkStampItemNamespace
+	case i.address.IsZero():
+		return nil, errMarshalInvalidChunkStampItemAddress
+	case i.stamp == nil:
+		return nil, errMarshalInvalidChunkStampItemStamp
+	}
+
+	buf := make([]byte, 8+len(i.namespace)+postage.StampSize)
+
+	l := 0
+	binary.LittleEndian.PutUint64(buf[l:l+8], uint64(len(i.namespace)))
+	l += 8
+	copy(buf[l:l+len(i.namespace)], i.namespace)
+	l += len(i.namespace)
+	data, err := i.stamp.MarshalBinary()
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal chunkstamp.item: %w", err)
+	}
+	copy(buf[l:l+postage.StampSize], data)
+
+	return buf, nil
+}
+
+// Unmarshal implements the storage.Item interface.
+func (i *item) Unmarshal(bytes []byte) error {
+	if len(bytes) < 8 {
+		return errUnmarshalInvalidChunkStampItemSize
+	}
+	nsLen := int(binary.LittleEndian.Uint64(bytes))
+	if len(bytes) != 8+nsLen+postage.StampSize {
+		return errUnmarshalInvalidChunkStampItemSize
+	}
+
+	// Ensure that the address is set already in the item.
+	if i.address.IsZero() {
+		return errUnmarshalInvalidChunkStampItemAddress
+	}
+
+	ni := &item{address: i.address.Clone()}
+	l := 8
+	ni.namespace = append(make([]byte, 0, nsLen), bytes[l:l+nsLen]...)
+	l += nsLen
+	stamp := new(postage.Stamp)
+	if err := stamp.UnmarshalBinary(bytes[l : l+postage.StampSize]); err != nil {
+		if errors.Is(err, postage.ErrStampInvalid) {
+			return errUnmarshalInvalidChunkStampItemSize
+		}
+		return fmt.Errorf("unable to unmarshal chunkstamp.item: %w", err)
+	}
+	ni.stamp = stamp
+	*i = *ni
+	return nil
+}
+
+// Clone implements the storage.Item interface.
+func (i *item) Clone() storage.Item {
+	if i == nil {
+		return nil
+	}
+	clone := &item{
+		namespace: append([]byte(nil), i.namespace...),
+		address:   i.address.Clone(),
+	}
+	if i.stamp != nil {
+		clone.stamp = i.stamp.Clone()
+	}
+	return clone
+}
+
+// String implements the storage.Item interface.
+func (i item) String() string {
+	return path.Join(i.Namespace(), i.ID())
+}
+
+// Load returns swarm.Stamp related to the given address and batchID.
+// The storage.ErrNoStampsForChunk is returned if no record is found.
+func Load(s internal.Storage, namespace string, addr swarm.Address, batchID []byte) (swarm.Stamp, error) {
```

Comment:
Now that we have a separate package for this would be it be better to have two functions?

Load with batch ID and LoadAny for getting any stamp.

Reply:
Done.